cura/Settings/ContainerManager.py

```diff
@@ -350,6 +351,194 @@ def importContainer(self, file_url):
+        stacks.insert(0, global_stack)
+
+        for stack in stacks:
+            user_container = stack.getTop()
+            quality_container = stack.findContainer(type = "quality")
+            quality_changes_container = stack.findContainer(type = "quality_changes")
+            if not quality_container or not quality_changes_container:
+                UM.Logger.log("w", "No quality or quality changes container found in stack %s, ignoring it", stack.getId())
+                return False
+
+            new_quality_changes = user_container.duplicate(stack.getId() + "_" + unique_id, unique_name)
+            new_quality_changes.setMetaDataEntry("type", "quality_changes")
+            new_quality_changes.addMetaDataEntry("quality", quality_container.getMetaDataEntry("quality_type"))
+
+            if not global_stack.getMetaDataEntry("has_machine_quality"):
+                new_quality_changes.setDefinition(UM.Settings.ContainerRegistry.getInstance().findContainers(id = "fdmprinter")[0])
+
+            if global_stack.getMetaDataEntry("has_materials"):
+                material = stack.findContainer(type = "material")
+                new_quality_changes.addMetaDataEntry("material", material.getId())
+
+            UM.Settings.ContainerRegistry.getInstance().addContainer(new_quality_changes)
+
+            stack.replaceContainer(stack.getContainerIndex(quality_changes_container), new_quality_changes)
+            stack.getTop().clear()
+
+        UM.Application.getInstance().getMachineManager().activeQualityChanged.emit()
+        return True
+
+    @pyqtSlot(str, result = bool)
+    def removeQualityChanges(self, quality_name):
+        if not quality_name:
+            return False
+
+        for container in self._getQualityContainers(quality_name):
+            UM.Settings.ContainerRegistry.getInstance().removeContainer(container.getId())
+
+        return True
+
+    @pyqtSlot(str, str, result = bool)
+    def renameQualityChanges(self, quality_name, new_name):
+        if not quality_name or not new_name:
+            return False
+
+        if quality_name == new_name:
+            return True
+
+        global_stack = UM.Application.getInstance().getGlobalContainerStack()
+        if not global_stack:
+            return False
+
+        UM.Application.getInstance().getMachineManager().blurSettings.emit()
+
+        new_name = UM.Settings.ContainerRegistry.getInstance().uniqueName(new_name)
+        new_id = new_name.lower()
+        new_id.replace(" ", "_")
+
+        for container in self._getQualityContainers(quality_name):
+            stack_id = container.getMetaDataEntry("extruder", global_stack.getId())
+
+            UM.Settings.ContainerRegistry.getInstance().renameContainer(container.getId(), new_name, stack_id + "_" + new_id)
+
+        UM.Application.getInstance().getMachineManager().activeQualityChanged.emit()
+        return True
+
+    @pyqtSlot(str, result = str)
+    def duplicateQuality(self, quality_name):
```

Comment:
No. It creates quality-changes containers. However, it accepts both a quality-changes container name or a quality container name. For quality changes, it can simply duplicate them. For quality, it creates a new set of quality changes based on the supplied quality.